drainer/translator/translator.go

```diff
@@ -158,11 +160,11 @@ func DecodeOldAndNewRow(b []byte, cols map[int64]*model.ColumnInfo, loc *time.Lo
 	parsedCols := cnt / 2
 	isInvalid := len(newRow) != len(oldRow) || (len(cols) != parsedCols && len(cols)-1 != parsedCols)
 	if isInvalid {
-		return nil, nil, errors.Errorf("row data is corrupted %v", b)
+		return nil, nil, errors.Errorf("row data is corrupted cols num: %d, oldRow: %v, newRow: %v", len(cols), oldRow, newRow)
 	}
 	if parsedCols == len(cols)-1 {
```

Comment:
It seems there may be multi columns changed, I tested this sample and drainer quit with error.

```sql
/* init */ drop table if exists accounts;
/* init */ create table accounts(id int primary key, balance bigint, aa int);
/* init */ insert into accounts values(1, 100, 1), (2, 0, 2), (3, 1, 3);

/* TXN1 */ set session tidb_enable_amend_pessimistic_txn = 1;
/* DDL */ set session tidb_enable_amend_pessimistic_txn = 1;
/* TXN1 */ begin;
/* DDL */ alter table accounts add column bb int default 10;
/* DDL */ alter table accounts add column cc int default 20;
/* TXN1 */ update accounts set balance = id where 1;
/* TXN1 */ commit;
/* TXN1 */ select * from accounts;
```

```
[2020/11/03 07:58:30.335 +00:00] [ERROR] [server.go:289] ["syncer exited abnormal"] [error="failed to add item: gen update fail: table `test`.`accounts`: row data is corrupted cols num: 5, oldRow: map[1:KindInt64 1 2:KindInt64 100 3:KindInt64 1], newRow: map[1:KindInt64 1 2:KindInt64 1 3:KindInt64 1]"] [errorVerbose="row data is corrupted cols num: 5, oldRow: map[1:KindInt64 1 2:KindInt64 100 3:KindInt64 1], newRow: map[1:KindInt64 1 2:KindInt64 1 3:KindInt64 1]\ngithub.com/pingcap/tidb-binlog/drainer/translator.DecodeOldAndNewRow\n\t/home/tidb/issue-20500/tidb-binlog/drainer/translator/translator.go:163\ngithub.com/pingcap/tidb-binlog/drainer/translator.updateDecoder.decode\n\t/home/tidb/issue-20500/tidb-binlog/drainer/translator/translator.go:211\ngithub.com/pingcap/tidb-binlog/drainer/translator.genMysqlUpdate\n\t/home/tidb/issue-20500/tidb-binlog/drainer/translator/mysql.go:65\ngithub.com/pingcap/tidb-binlog/drainer/translator.TiBinlogToTxn\n\t/home/tidb/issue-20500/tidb-binlog/drainer/translator/mysql.go:159\ngithub.com/pingcap/tidb-binlog/drainer/sync.(*MysqlSyncer).Sync\n\t/home/tidb/issue-20500/tidb-binlog/drainer/sync/mysql.go:186\ngithub.com/pingcap/tidb-binlog/drainer.(*Syncer).run\n\t/home/tidb/issue-20500/tidb-binlog/drainer/syncer.go:378\ngithub.com/pingcap/tidb-binlog/drainer.(*Syncer).Start\n\t/home/tidb/issue-20500/tidb-binlog/drainer/syncer.go:132\ngithub.com/pingcap/tidb-binlog/drainer.(*Server).Start.func4\n\t/home/tidb/issue-20500/tidb-binlog/drainer/server.go:288\ngithub.com/pingcap/tidb-binlog/drainer.(*taskGroup).start.func1\n\t/home/tidb/issue-20500/tidb-binlog/drainer/util.go:75\nruntime.goexit\n\t/usr/local/share/go/src/runtime/asm_amd64.s:1374\ntable `test`.`accounts`\ngen update fail\nfailed to add item"]
```